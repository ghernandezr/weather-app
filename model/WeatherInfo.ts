import Temperature from "./Temperature";

interface WeatherInfo {
  city?: string;
  icon?: string;
  latitude?: number;
  longitude?: number;
  temperatures?: Temperature[];
}

export default WeatherInfo;
