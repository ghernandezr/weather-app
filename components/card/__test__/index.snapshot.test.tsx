import renderer from "react-test-renderer";
import Card from "..";
import WeatherInfo from "../../../model/WeatherInfo";

const weather: WeatherInfo = {
  city: "Miami",
  icon: "",
  id: 1234,
  latitude: 12.5,
  longitude: 14.45,
  temperatures: [],
};

it("renders correctly", () => {
  const tree = renderer.create(<Card weather={weather} />).toJSON();
  expect(tree).toMatchSnapshot();
});
