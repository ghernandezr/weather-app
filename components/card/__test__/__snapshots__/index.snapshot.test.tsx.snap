// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="group  rounded-3xl w-[500px] h-[425px] transition-colors  ease-in-out delay-75 bg-gradient-to-r from-[#000000] to-[#023047]  text-white"
>
  <div
    className="flex items-center justify-between px-9 py-9"
  >
    <i
      className="wi  transition-colors ease-in-out delay-75 text-4xl sm:text-6xl text-[#8ECAE6]"
    />
    <div>
      <h1
        className="text-xl sm:text-[32px]"
      >
        Miami
      </h1>
      <p
        className="text-sm sm:text-2xl mt-3"
      >
        12.5
        ° N, 
        14.45
        ° W
      </p>
    </div>
  </div>
  <div
    className="flex px-4 md:px-8 justify-between items-center mt-16"
  />
  <div
    className="flex items-center justify-center mt-16 "
  >
    <svg
      aria-hidden="true"
      className="h-9 w-9 hidden cursor-pointer group-hover:block"
      fill="none"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth={1.5}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M14.74 9l-.346 9m-4.788 0L9.26 9m9.968-3.21c.342.052.682.107 1.022.166m-1.022-.165L18.16 19.673a2.25 2.25 0 01-2.244 2.077H8.084a2.25 2.25 0 01-2.244-2.077L4.772 5.79m14.456 0a48.108 48.108 0 00-3.478-.397m-12 .562c.34-.059.68-.114 1.022-.165m0 0a48.11 48.11 0 013.478-.397m7.5 0v-.916c0-1.18-.91-2.164-2.09-2.201a51.964 51.964 0 00-3.32 0c-1.18.037-2.09 1.022-2.09 2.201v.916m7.5 0a48.667 48.667 0 00-7.5 0"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  </div>
</div>
`;
